File: Algorithm/Data_Structure/Bit_Operations/1486_easy.ts
/**
 * @Time 2024/10/01 23:40
 * @Author  : https://github.com/hakusai22
 * @题目     :
 * @参考     :
 * @时间复杂度:
 * @空间复杂度:
 */

function xorOperation(n: number, start: number): number {
    let ans = 0;
    for (let i = 0; i < n; i++) {
        ans ^= (start + 2 * i);
    }
    return ans;
};

File: Algorithm/Data_Structure/Bit_Operations/1486_easy.go
package Bit_Operations

/*
    @Author  : https://github.com/hakusai22
    @Time    : 2024/10/01 15:39
    @题目     : https://leetcode.cn/problems/xor-operation-in-an-array/
    @参考     :
    @时间复杂度:
    @空间复杂度:

 数据范围:

*/

func xorOperation(n int, start int) int {
	ans := start
	for i := 1; i < n; i++ {
		ans ^= start + 2*i
	}
	return ans
}
